src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/BookingTransformer.kt

```diff
@@ -72,12 +73,20 @@ class BookingTransformer(
     else -> throw RuntimeException("Could not determine status for Booking ${jpa.id}")
   }
 
-  private fun determineTemporaryAccommodationStatus(jpa: BookingEntity) = when {
-    jpa.cancellation != null -> BookingStatus.cancelled
-    jpa.departure != null -> BookingStatus.departed
-    jpa.arrival != null -> BookingStatus.arrived
-    jpa.nonArrival != null -> BookingStatus.notMinusArrived
-    jpa.confirmation != null -> BookingStatus.confirmed
-    else -> BookingStatus.provisional
+  private fun determineTemporaryAccommodationStatus(jpa: BookingEntity): BookingStatus {
+    val hasNonZeroDayTurnaround = jpa.turnaround != null && jpa.turnaround!!.workingDayCount != 0
+    val hasZeroDayTurnaround = jpa.turnaround != null && jpa.turnaround!!.workingDayCount == 0
```

Comment:
There's a small window of time between the migration that gave turnarounds to existing bookings and the automatic creation of turnarounds for new bookings where bookings might not have a turnaround, as what was originally going to be one PR was split into two to unblock some work. In this case, I'd expect it to act as if it did have a turnaround of zero days, but as far as I can tell this situation would be incorrectly reported as a provisional booking.

Reply:
Good point - have accounted for this by treating absense of turnaround as the same as a zero day one